src/common/address-book/actions.js

```diff
@@ -0,0 +1,66 @@
+import * as types from './types';
+import { createAliasedAction } from 'electron-redux';
+import { getGlobalContext } from '../context';
+
+const addEntry = async entry => {
```

Comment:
- async action should be an operation (like any non atomic action). Atomic actions always update redux store in a predictable way.
- shouldn't async action return a thunk like this?:
``` entry => async (dispatch, getState) => { ```

Reply:
@sk91 I agree with you but here it is a different case. We need this to run on the main process otherwise it will error out on knex. I tried doing the other around but if I run this code on operations it will error out. Cheers